src/hotspot/share/nmt/memTracker.hpp

```diff
@@ -82,18 +83,21 @@ class MemTracker : AllStatic {
   }
 
   static inline void* record_malloc(void* mem_base, size_t size, MemTag mem_tag,
-    const NativeCallStack& stack) {
+    const NativeCallStack& stack, void* old_base = nullptr) {
     assert(mem_base != nullptr, "caller should handle null");
+    void* ptr = mem_base;
     if (enabled()) {
-      return MallocTracker::record_malloc(mem_base, size, mem_tag, stack);
+      ptr = MallocTracker::record_malloc(mem_base, size, mem_tag, stack);
     }
-    return mem_base;
+    NMT_MemoryLogRecorder::record_alloc(mem_tag, size, mem_base, &stack, old_base);
+    return ptr;
   }
```

Comment:
I'm not a fan of making `record_malloc` aware of the fact that it may be used within a `realloc` context with the addition of the `old_base`. I'd like to see this parameter removed.